functions/_pure_prompt_symbol.fish

```diff
@@ -14,5 +17,18 @@ function _pure_prompt_symbol \
         end
     end
 
-    echo "$color_symbol$pure_symbol"
+    if test reverse_symbol_instead_of_default_mode_prompt
+      # Do nothing if not in vi mode
+      if test "$fish_key_bindings" = "fish_vi_key_bindings"
+          or test "$fish_key_bindings" = "fish_hybrid_key_bindings"
```

Comment:
You should use `string match` and glob/regex to test the key_binding mode 

    if test (string match fish_{vi,hybrid}_key_bindings $fish_key_bindings)

or to be more explicit:

    set --local is_vi_mode (string match fish_{vi,hybrid}_key_bindings $fish_key_bindings)
    if test is_vi_mode